### DISCUSSION TITLE
Will UI class works with physics?

### DISCUSSION THREAD

**Discussion by crapthings** [rexrainbow/phaser3-rex-notes#230] [Q&A]
![image](https://user-images.githubusercontent.com/1147704/150125067-ebf1b088-10af-4f19-8f90-e7263272f53e.png)

collide doesn't work

```js
import _ from 'lodash'
import Phaser from 'phaser'
import { Label } from 'phaser3-rex-plugins/templates/ui/ui-components.js'
import CircleMaskImage from 'phaser3-rex-plugins/plugins/circlemaskimage.js'

import createPlayerBehavior from '../behaviors/player'

export default class extends CircleMaskImage {
  constructor (scene, x, y, key, frame, config) {
    super(scene, x, y, key, frame, config)
    scene.physics.add.existing(this)
    scene.add.existing(this)
    this.create()
  }

  create = () => {
    this.playerBehavior = createPlayerBehavior(this)
  }

  preUpdate () {
    this.playerBehavior.step()
  }

  seek = () => {
    const enemies = this.scene.enemies

    if (!enemies) return

    const targetEnemy = this.getClosest(enemies.getChildren())

    if (!targetEnemy) {
      this.scene.scene.restart()
      return
    }

    this.moveToTarget(targetEnemy)
  }

  moveToTarget = (target, speed = 100) => {
    this.scene.physics.moveToObject(this, target, speed)
  }

  getClosest = (targets) => {
    return this.scene.physics.closest(this, targets)
  }

  getDistance = (target) => {
    return Phaser.Math.Distance.BetweenPoints(this, target)
  }
}
```

and I want make a circle around avatar that represent player's hp, but it doesn't move together.

should I wrap rexui with a container? and move that container?

![image](https://user-images.githubusercontent.com/1147704/150125658-742dd41b-19cc-4963-9f39-6a42182ea0ac.png)

what is the best practice to make such UI with rexUI

![image](https://user-images.githubusercontent.com/1147704/150126139-953a3db1-40f5-4219-aad6-f42da7f585d9.png)

1. a image with circle mask
2. maybe a border(stroke) with the mask
3. serval label text?
4. fx around the object?

**Comment by rexrainbow** [comment]
Any rexui sizer is extended from [containerLite](https://rexrainbow.github.io/phaser3-rex-notes/docs/site/containerlite/), moves the topmost ui object will also moves children. (Moving child directly is another story).
It is fine to add a rexui sizer to a built-in container ([reference](https://rexrainbow.github.io/phaser3-rex-notes/docs/site/containerlite/#add-to-container)), if this rexui sizer does not have mask (GridTable, ScrollablePanel, TextArea have mask)

CircleMaskImage does not have mask, it draws circular crop image on a canvas.

**Reply by crapthings** [reply]
will sizer works with scene.physics.add.existing(this)

anyway i'll try it, thanks @rexrainbow

**Comment by rexrainbow** [comment]
Yes, it can. You can set position directly, or by tween, or with physics engine, ideally.

But I did not try physics engine before.
